README.md
# ELS - Enterprise Linux/EuroLinux Sources - brasero
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
# Generated by Makefile. Do not edit.

commit 898d7473a47e60acedbf36e5c178c180ac521e33
Author: Matthias Clasen <mclasen@redhat.com>
Date:   Mon Mar 25 22:45:08 2013 -0400

    3.8.0

 NEWS         | 4 ++++
 configure.ac | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

commit 52c0e5aed11ef2fd6df70b21f068270556d2c318
Author: Alexandre Franke <alexandre.franke@gmail.com>
Date:   Mon Mar 25 23:35:03 2013 +0100

    Update French translation

 help/fr/fr.po | 55 +++++++++++++++++++++++++++++++++++++++----------------
 1 file changed, 39 insertions(+), 16 deletions(-)

commit d3d6acb107dc08958b814c37d1f147ae2a58d9d9
Author: Shankar Prasad <svenkate@redhat.com>
Date:   Mon Mar 25 23:09:51 2013 +0530

    Updated kn translations

 po/kn.po | 613 ++++++++++++++++++++++++++++++++-------------------------------
 1 file changed, 310 insertions(+), 303 deletions(-)

commit f406837ddf20fdcccf44d9c8664832843026f75c
Author: Daniel Mustieles <daniel.mustieles@gmail.com>
Date:   Mon Mar 25 17:44:38 2013 +0100

    Updated Spanish translation

 help/es/es.po | 52 ++++++++++++++++++++++++++++++++++++----------------
 1 file changed, 36 insertions(+), 16 deletions(-)

commit a04c9d99a1791b6d9fd19b698a279d3fcde6e737
Author: Krishnababu Krothapalli <kkrothap@redhat.com>
Date:   Mon Mar 25 14:55:28 2013 +0530

    Updated Telugu Translations

 po/te.po | 939 ++++++++++++++++++++++++++++++++++++---------------------------
 1 file changed, 536 insertions(+), 403 deletions(-)

commit 45182b91036a25a099e344692a9efc6ea13f8cc4
Author: Piotr Drąg <piotrdrag@gmail.com>
Date:   Sun Mar 24 19:06:59 2013 +0100

    Fix translations of keywords in .desktop files

 po/el.po | 2 +-
 po/ml.po | 2 +-
 po/pt.po | 2 +-
 po/sk.po | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

commit 38ed48f1889983fabdd9e3eb7eb77eee167fadb0
Author: Mario Blättermann <mario.blaettermann@gmail.com>
Date:   Sun Mar 24 13:10:41 2013 +0100

    [l10n] Updated German doc translation

 help/de/de.po | 46 +++++++++++++++++++++++++++++++++-------------
 1 file changed, 33 insertions(+), 13 deletions(-)

commit c04f0c3dd16ad8cca9036ed0f9a61e067c8147f3
Author: Eliane Ramos Pereira <elirpereira@gmail.com>
Date:   Wed Aug 15 01:29:30 2012 -0300

    Fix mistake in user help, bug 669390

 help/C/project-save.page | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

commit 58163e550405ba5267a751ca5f0b09d23b7710a1
Author: Seong-ho Cho <darkcircle.0426@gmail.com>
Date:   Fri Mar 22 23:19:00 2013 +0900

    Updated Korean document translation

 help/ko/ko.po | 100 +++++++++++++++++++++++++++++-----------------------------
 1 file changed, 50 insertions(+), 50 deletions(-)

commit b9daad784a1d69888c7c5fcdd7c98d778db16e9e
Author: Ani Peter <apeter@redhat.com>
Date:   Fri Mar 22 11:27:51 2013 +0530

    Completed brasero for Malayalam

 po/ml.po | 485 +++++++++++++++++++++++++++++++++++++++++++--------------------
 1 file changed, 331 insertions(+), 154 deletions(-)

commit 3ac1b69162c90c70bf0eab839af2c1bd4d4e2bce
Author: Seong-ho Cho <darkcircle.0426@gmail.com>
Date:   Fri Mar 22 04:37:33 2013 +0900

    Updated Korean document translation

 help/ko/ko.po | 1207 +++++++++++++++++++++++++++++++++------------------------
 1 file changed, 711 insertions(+), 496 deletions(-)

commit 2a199f2a6197f6cf52d6506add2483968e92152a
Author: Rajesh Ranjan <rranjan@redhat.com>
Date:   Thu Mar 21 12:10:46 2013 +0530

    hindi translation

 po/hi.po | 425 +++++++++++++++++++++++++++++++++------------------------------
 1 file changed, 220 insertions(+), 205 deletions(-)

commit d86d98aec5f55280209712071e4836b0095c12b3
Author: ManojKumar Giri <mgiri@redhat.com>
Date:   Wed Mar 20 16:56:37 2013 +0530

    Updated Odia Translation.

 po/or.po | 615 ++++++++++++++++++++++++++++++++-------------------------------
 1 file changed, 311 insertions(+), 304 deletions(-)

commit 7a49bf77d0e243bd8025e369a7449143229e6588
Author: Victor Ibragimov <victor.ibragimov@gmail.com>
Date:   Tue Mar 19 21:02:42 2013 +0100

    [l10n] Added Tadjik translation

 po/LINGUAS |    1 +
 po/tg.po   | 4795 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 4796 insertions(+)

commit 8803acbbdb898a6e3cbdc9a53abe26c9bd15fcd0
Author: Pavol Klačanský <pavol@klacansky.com>
Date:   Mon Mar 18 20:57:02 2013 +0100

    Updated slovak translation

 po/sk.po | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

commit 0e3d55aa266e8cd360f3fd25f0ecba5fe04d7440
Author: Sandeep Sheshrao Shedmake <sshedmak@redhat.com>
Date:   Mon Mar 18 12:52:17 2013 +0530

    Updated Marathi Translations

 po/mr.po | 623 ++++++++++++++++++++++++++++++++-------------------------------
 1 file changed, 316 insertions(+), 307 deletions(-)

commit cdc7a2ba8aa6effefde3a75025ebabda71396b90
Author: Enrico Nicoletto <liverig@gmail.com>
Date:   Sun Mar 17 17:50:52 2013 -0300

    Updated Brazilian Portuguese translation for help files

 help/pt_BR/pt_BR.po | 311 ++++++++++++++++++++++++++++++++++++++++------------
 1 file changed, 240 insertions(+), 71 deletions(-)

commit 2544800b2a9d5c7396f2de8cc650f48f6bd1576b
Author: Rafael Ferreira <rafael.f.f1@gmail.com>
Date:   Sun Mar 17 17:49:54 2013 -0300

    Updated Brazilian Portuguese translation

 po/pt_BR.po | 78 ++++++++++++++++++++++++++++++-------------------------------
 1 file changed, 39 insertions(+), 39 deletions(-)

commit 558697627e42078a9e2ba21acb199872fd9f2ea1
Author: Carles Ferrando <carles.ferrando@gmail.com>
Date:   Sun Mar 17 19:57:12 2013 +0100

    [l10n] Updated Catalan (Valencian) translation

 po/ca@valencia.po | 665 ++++++++++++++++++++++++++++--------------------------
 1 file changed, 339 insertions(+), 326 deletions(-)

commit 868718145bedba1d83ba76a90ad9c57e5c77c606
Author: A S Alam <aalam@users.sf.net>
Date:   Sun Mar 17 19:40:32 2013 +0530

    Punjabi: Translation updated (aalam)

 po/pa.po | 310 ++++++++++++++++++++++++++++++++-------------------------------

src/libbrasero-burn/brasero-burn-lib.h
/* -*- Mode: C; indent-tabs-mode: t; c-basic-offset: 8; tab-width: 8 -*- */
/*
 * Libbrasero-burn
 * Copyright (C) Philippe Rouquier 2005-2009 <bonfire-app@wanadoo.fr>
 *
 * Libbrasero-burn is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * The Libbrasero-burn authors hereby grant permission for non-GPL compatible
 * GStreamer plugins to be used and distributed together with GStreamer
 * and Libbrasero-burn. This permission is above and beyond the permissions granted
 * by the GPL license by which Libbrasero-burn is covered. If you modify this code
 * you may extend this exception to your version of the code, but you are not
 * obligated to do so. If you do not wish to do so, delete this exception
 * statement from your version.
 * 
 * Libbrasero-burn is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Library General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to:
 * 	The Free Software Foundation, Inc.,
 * 	51 Franklin Street, Fifth Floor
 * 	Boston, MA  02110-1301, USA.
 */

#ifndef _BRASERO_BURN_LIB_
#define _BRASERO_BURN_LIB_

#include <glib.h>

#include <brasero-enums.h>
#include <brasero-error.h>
#include <brasero-track-type.h>

G_BEGIN_DECLS

/**
 * Some needed information about the library
 */

#define LIBBRASERO_BURN_VERSION_MAJOR						\
	3
#define LIBBRASERO_BURN_VERSION_MINOR						\
	6
#define LIBBRASERO_BURN_VERSION_MICRO						\
	0
#define LIBBRASERO_BURN_AGE							\
	332


/**
 * Function to start and stop the library
 */

gboolean
brasero_burn_library_start (int *argc,
                            char **argv []);

void
brasero_burn_library_stop (void);

/**
 * GOptionGroup for apps
 */

GOptionGroup *
brasero_burn_library_get_option_group (void);

/**
 * Allows to get some information about capabilities
 */

GSList *
brasero_burn_library_get_plugins_list (void);

gboolean
brasero_burn_library_can_checksum (void);

BraseroBurnResult
brasero_burn_library_input_supported (BraseroTrackType *type);

BraseroMedia
brasero_burn_library_get_media_capabilities (BraseroMedia media);

G_END_DECLS

#endif /* _BRASERO_BURN_LIB_ */



src/libbrasero-burn/libbrasero-marshal.h

#ifndef __brasero_marshal_MARSHAL_H__
#define __brasero_marshal_MARSHAL_H__

#include	<glib-object.h>

G_BEGIN_DECLS

/* INT:VOID (./libbrasero-marshal.list:1) */
extern void brasero_marshal_INT__VOID (GClosure     *closure,
                                       GValue       *return_value,
                                       guint         n_param_values,
                                       const GValue *param_values,
                                       gpointer      invocation_hint,
                                       gpointer      marshal_data);

/* INT:INT (./libbrasero-marshal.list:2) */
extern void brasero_marshal_INT__INT (GClosure     *closure,
                                      GValue       *return_value,
                                      guint         n_param_values,
                                      const GValue *param_values,
                                      gpointer      invocation_hint,
                                      gpointer      marshal_data);

/* INT:STRING (./libbrasero-marshal.list:3) */
extern void brasero_marshal_INT__STRING (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
                                         const GValue *param_values,
                                         gpointer      invocation_hint,
                                         gpointer      marshal_data);

/* INT:OBJECT,INT,INT (./libbrasero-marshal.list:4) */
extern void brasero_marshal_INT__OBJECT_INT_INT (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
                                                 const GValue *param_values,
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);

/* INT:POINTER,BOOLEAN (./libbrasero-marshal.list:5) */
extern void brasero_marshal_INT__POINTER_BOOLEAN (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
                                                  const GValue *param_values,
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);

/* BOOLEAN:VOID (./libbrasero-marshal.list:6) */
extern void brasero_marshal_BOOLEAN__VOID (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
                                           const GValue *param_values,
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);

/* BOOLEAN:STRING (./libbrasero-marshal.list:7) */
extern void brasero_marshal_BOOLEAN__STRING (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
                                             const GValue *param_values,
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);

/* BOOLEAN:POINTER (./libbrasero-marshal.list:8) */
extern void brasero_marshal_BOOLEAN__POINTER (GClosure     *closure,
                                              GValue       *return_value,
                                              guint         n_param_values,
                                              const GValue *param_values,
                                              gpointer      invocation_hint,
                                              gpointer      marshal_data);

/* VOID:INT,STRING (./libbrasero-marshal.list:9) */
extern void brasero_marshal_VOID__INT_STRING (GClosure     *closure,
                                              GValue       *return_value,
                                              guint         n_param_values,
                                              const GValue *param_values,
                                              gpointer      invocation_hint,
                                              gpointer      marshal_data);

/* VOID:POINTER,STRING (./libbrasero-marshal.list:10) */
extern void brasero_marshal_VOID__POINTER_STRING (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
                                                  const GValue *param_values,
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);

/* VOID:POINTER,POINTER (./libbrasero-marshal.list:11) */
extern void brasero_marshal_VOID__POINTER_POINTER (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
                                                   const GValue *param_values,
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);

/* VOID:OBJECT,BOOLEAN (./libbrasero-marshal.list:12) */
extern void brasero_marshal_VOID__OBJECT_BOOLEAN (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
                                                  const GValue *param_values,
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);

/* VOID:OBJECT,UINT (./libbrasero-marshal.list:13) */
extern void brasero_marshal_VOID__OBJECT_UINT (GClosure     *closure,
                                               GValue       *return_value,
                                               guint         n_param_values,
                                               const GValue *param_values,
                                               gpointer      invocation_hint,
                                               gpointer      marshal_data);

/* VOID:BOOLEAN,BOOLEAN (./libbrasero-marshal.list:14) */
extern void brasero_marshal_VOID__BOOLEAN_BOOLEAN (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
                                                   const GValue *param_values,
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);

/* VOID:DOUBLE,DOUBLE,LONG (./libbrasero-marshal.list:15) */
extern void brasero_marshal_VOID__DOUBLE_DOUBLE_LONG (GClosure     *closure,
                                                      GValue       *return_value,
                                                      guint         n_param_values,
                                                      const GValue *param_values,
                                                      gpointer      invocation_hint,
                                                      gpointer      marshal_data);

/* VOID:POINTER,UINT,POINTER (./libbrasero-marshal.list:16) */
extern void brasero_marshal_VOID__POINTER_UINT_POINTER (GClosure     *closure,
                                                        GValue       *return_value,
                                                        guint         n_param_values,
                                                        const GValue *param_values,
                                                        gpointer      invocation_hint,
                                                        gpointer      marshal_data);

G_END_DECLS

#endif /* __brasero_marshal_MARSHAL_H__ */



src/libbrasero-media/libbrasero-marshal.c

#ifndef __brasero_marshal_MARSHAL_H__
#define __brasero_marshal_MARSHAL_H__

#include	<glib-object.h>

G_BEGIN_DECLS

#ifdef G_ENABLE_DEBUG
#define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
#define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
#define g_marshal_value_peek_int(v)      g_value_get_int (v)
#define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
#define g_marshal_value_peek_long(v)     g_value_get_long (v)
#define g_marshal_value_peek_ulong(v)    g_value_get_ulong (v)
#define g_marshal_value_peek_int64(v)    g_value_get_int64 (v)
#define g_marshal_value_peek_uint64(v)   g_value_get_uint64 (v)
#define g_marshal_value_peek_enum(v)     g_value_get_enum (v)
#define g_marshal_value_peek_flags(v)    g_value_get_flags (v)
#define g_marshal_value_peek_float(v)    g_value_get_float (v)
#define g_marshal_value_peek_double(v)   g_value_get_double (v)
#define g_marshal_value_peek_string(v)   (char*) g_value_get_string (v)
#define g_marshal_value_peek_param(v)    g_value_get_param (v)
#define g_marshal_value_peek_boxed(v)    g_value_get_boxed (v)
#define g_marshal_value_peek_pointer(v)  g_value_get_pointer (v)
#define g_marshal_value_peek_object(v)   g_value_get_object (v)
#define g_marshal_value_peek_variant(v)  g_value_get_variant (v)
#else /* !G_ENABLE_DEBUG */
/* WARNING: This code accesses GValues directly, which is UNSUPPORTED API.
 *          Do not access GValues directly in your code. Instead, use the
 *          g_value_get_*() functions
 */
#define g_marshal_value_peek_boolean(v)  (v)->data[0].v_int
#define g_marshal_value_peek_char(v)     (v)->data[0].v_int
#define g_marshal_value_peek_uchar(v)    (v)->data[0].v_uint
#define g_marshal_value_peek_int(v)      (v)->data[0].v_int
#define g_marshal_value_peek_uint(v)     (v)->data[0].v_uint
#define g_marshal_value_peek_long(v)     (v)->data[0].v_long
#define g_marshal_value_peek_ulong(v)    (v)->data[0].v_ulong
#define g_marshal_value_peek_int64(v)    (v)->data[0].v_int64
#define g_marshal_value_peek_uint64(v)   (v)->data[0].v_uint64
#define g_marshal_value_peek_enum(v)     (v)->data[0].v_long
#define g_marshal_value_peek_flags(v)    (v)->data[0].v_ulong
#define g_marshal_value_peek_float(v)    (v)->data[0].v_float
#define g_marshal_value_peek_double(v)   (v)->data[0].v_double
#define g_marshal_value_peek_string(v)   (v)->data[0].v_pointer
#define g_marshal_value_peek_param(v)    (v)->data[0].v_pointer
#define g_marshal_value_peek_boxed(v)    (v)->data[0].v_pointer
#define g_marshal_value_peek_pointer(v)  (v)->data[0].v_pointer
#define g_marshal_value_peek_object(v)   (v)->data[0].v_pointer
#define g_marshal_value_peek_variant(v)  (v)->data[0].v_pointer
#endif /* !G_ENABLE_DEBUG */


/* VOID:STRING,STRING (./libbrasero-marshal.list:1) */
extern void brasero_marshal_VOID__STRING_STRING (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
                                                 const GValue *param_values,
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);
void
brasero_marshal_VOID__STRING_STRING (GClosure     *closure,
                                     GValue       *return_value G_GNUC_UNUSED,
                                     guint         n_param_values,
                                     const GValue *param_values,
                                     gpointer      invocation_hint G_GNUC_UNUSED,
                                     gpointer      marshal_data)
{
  typedef void (*GMarshalFunc_VOID__STRING_STRING) (gpointer     data1,
                                                    gpointer     arg_1,
                                                    gpointer     arg_2,
                                                    gpointer     data2);
  GMarshalFunc_VOID__STRING_STRING callback;
  GCClosure *cc = (GCClosure*) closure;
  gpointer data1, data2;

  g_return_if_fail (n_param_values == 3);

  if (G_CCLOSURE_SWAP_DATA (closure))
    {
      data1 = closure->data;
      data2 = g_value_peek_pointer (param_values + 0);
    }
  else
    {
      data1 = g_value_peek_pointer (param_values + 0);
      data2 = closure->data;
    }
  callback = (GMarshalFunc_VOID__STRING_STRING) (marshal_data ? marshal_data : cc->callback);

  callback (data1,
            g_marshal_value_peek_string (param_values + 1),
            g_marshal_value_peek_string (param_values + 2),
            data2);
}

G_END_DECLS

#endif /* __brasero_marshal_MARSHAL_H__ */



src/libbrasero-media/brasero-media.h
/* -*- Mode: C; indent-tabs-mode: t; c-basic-offset: 8; tab-width: 8 -*- */
/*
 * Libbrasero-media
 * Copyright (C) Philippe Rouquier 2005-2009 <bonfire-app@wanadoo.fr>
 *
 * Libbrasero-media is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * The Libbrasero-media authors hereby grant permission for non-GPL compatible
 * GStreamer plugins to be used and distributed together with GStreamer
 * and Libbrasero-media. This permission is above and beyond the permissions granted
 * by the GPL license by which Libbrasero-media is covered. If you modify this code
 * you may extend this exception to your version of the code, but you are not
 * obligated to do so. If you do not wish to do so, delete this exception
 * statement from your version.
 * 
 * Libbrasero-media is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Library General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to:
 * 	The Free Software Foundation, Inc.,
 * 	51 Franklin Street, Fifth Floor
 * 	Boston, MA  02110-1301, USA.
 */
 
#ifndef _BURN_MEDIA_H
#define _BURN_MEDIA_H

#include <glib.h>

G_BEGIN_DECLS

/**
 * Some needed information about the library
 */

#define LIBBRASERO_MEDIA_VERSION_MAJOR						\
	3
#define LIBBRASERO_MEDIA_VERSION_MINOR						\
	6
#define LIBBRASERO_MEDIA_VERSION_MICRO						\
	0
#define LIBBRASERO_MEDIA_AGE							\
	332
#define LIBBRASERO_MEDIA_INSTALL							\
	"/home/mclasen/gnome"

/**
 * To start and stop the library
 */

void
brasero_media_library_start (void);

void
brasero_media_library_stop (void);

GOptionGroup *
brasero_media_get_option_group (void);

/**
 * Errors
 */

GQuark brasero_media_quark (void);

/**
 * BRASERO_MEDIA_ERROR:
 *
 * The GQuark used for BraseroMediaError
 */
#define BRASERO_MEDIA_ERROR brasero_media_quark()

/**
 * BraseroMediaError:
 *
 * Error codes returned by libbrasero-media
 *
 */
typedef enum {
	BRASERO_MEDIA_ERROR_NONE,
	BRASERO_MEDIA_ERROR_GENERAL,
	BRASERO_MEDIA_ERROR_IMAGE_INVALID,
} BraseroMediaError;

/**
 * BraseroMedia:
 *
 * This enum allows to define all characteristics for a media (closed, writable, ...)
 * and identify all types of optical media.
 *
 */
typedef enum {
	BRASERO_MEDIUM_UNSUPPORTED		= -2,
	BRASERO_MEDIUM_BUSY			= -1,
	BRASERO_MEDIUM_NONE			= 0,

	/* types */
	BRASERO_MEDIUM_FILE			= 1,

	BRASERO_MEDIUM_CD			= 1 << 1,

	BRASERO_MEDIUM_DVD			= 1 << 2,

	BRASERO_MEDIUM_BD			= 1 << 3,

	/* subtype for DVD and BD types */
	BRASERO_MEDIUM_DUAL_L			= 1 << 4,

	/* DVD and DVD DL subtypes */
	BRASERO_MEDIUM_RAM			= 1 << 5,
	BRASERO_MEDIUM_PLUS			= 1 << 6,
	BRASERO_MEDIUM_SEQUENTIAL		= 1 << 7,
	BRASERO_MEDIUM_RESTRICTED		= 1 << 8,	/* DVD-RW only */

	/* DVD-R dual layer only subtype */
	BRASERO_MEDIUM_JUMP			= 1 << 9,

	/* BD subtypes */
	BRASERO_MEDIUM_RANDOM			= 1 << 10,
	BRASERO_MEDIUM_SRM			= 1 << 11,
	BRASERO_MEDIUM_POW			= 1 << 12,	/* This is Pseudo OverWrite */

	/* discs attributes */
	BRASERO_MEDIUM_REWRITABLE		= 1 << 14,
	BRASERO_MEDIUM_WRITABLE			= 1 << 15,
	BRASERO_MEDIUM_ROM			= 1 << 16,

	/* status of the disc */
	BRASERO_MEDIUM_BLANK			= 1 << 17,
	BRASERO_MEDIUM_CLOSED			= 1 << 18,
	BRASERO_MEDIUM_APPENDABLE		= 1 << 19,

	/* Only used for DVD+RW, DVD-RW restricted */
	BRASERO_MEDIUM_UNFORMATTED		= 1 << 20,

	BRASERO_MEDIUM_PROTECTED		= 1 << 21,
	BRASERO_MEDIUM_HAS_DATA			= 1 << 22,
	BRASERO_MEDIUM_HAS_AUDIO		= 1 << 23,
} BraseroMedia;

#define BRASERO_MEDIUM_CDROM		(BRASERO_MEDIUM_CD|		\
					 BRASERO_MEDIUM_ROM)
#define BRASERO_MEDIUM_CDR		(BRASERO_MEDIUM_CD|		\
					 BRASERO_MEDIUM_WRITABLE)
#define BRASERO_MEDIUM_CDRW		(BRASERO_MEDIUM_CD|		\
					 BRASERO_MEDIUM_REWRITABLE)
#define BRASERO_MEDIUM_DVD_ROM		(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_ROM)
#define BRASERO_MEDIUM_DVDR		(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_SEQUENTIAL|	\
					 BRASERO_MEDIUM_WRITABLE)
#define BRASERO_MEDIUM_DVDRW		(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_SEQUENTIAL|	\
					 BRASERO_MEDIUM_REWRITABLE)
#define BRASERO_MEDIUM_DVDRW_RESTRICTED	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_REWRITABLE|	\
					 BRASERO_MEDIUM_RESTRICTED)
#define BRASERO_MEDIUM_DVDR_DL		(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_DUAL_L|		\
					 BRASERO_MEDIUM_WRITABLE|	\
					 BRASERO_MEDIUM_SEQUENTIAL)
#define BRASERO_MEDIUM_DVDR_JUMP_DL	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_DUAL_L|		\
					 BRASERO_MEDIUM_WRITABLE|	\
					 BRASERO_MEDIUM_JUMP)
#define BRASERO_MEDIUM_DVDR_PLUS	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_WRITABLE|	\
					 BRASERO_MEDIUM_PLUS)
#define BRASERO_MEDIUM_DVDRW_PLUS	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_REWRITABLE|	\
					 BRASERO_MEDIUM_PLUS)
#define BRASERO_MEDIUM_DVDR_PLUS_DL	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_DUAL_L|		\
					 BRASERO_MEDIUM_WRITABLE|	\
					 BRASERO_MEDIUM_PLUS)
#define BRASERO_MEDIUM_DVDRW_PLUS_DL	(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_DUAL_L|		\
					 BRASERO_MEDIUM_REWRITABLE|	\
					 BRASERO_MEDIUM_PLUS)

#define BRASERO_MEDIUM_DVD_RAM		(BRASERO_MEDIUM_DVD|		\
					 BRASERO_MEDIUM_RAM)

/* BD types all exist in BD-R(E) DL as well */
#define BRASERO_MEDIUM_BD_ROM		(BRASERO_MEDIUM_BD|		\
					 BRASERO_MEDIUM_ROM)
#define BRASERO_MEDIUM_BDR_SRM		(BRASERO_MEDIUM_BD|		\
					 BRASERO_MEDIUM_SRM|		\
					 BRASERO_MEDIUM_WRITABLE)
#define BRASERO_MEDIUM_BDR_SRM_POW	(BRASERO_MEDIUM_BD|		\
					 BRASERO_MEDIUM_POW|		\
					 BRASERO_MEDIUM_SRM|		\
					 BRASERO_MEDIUM_WRITABLE)

/* This seems to be a really rare mode for BD-R */
#define BRASERO_MEDIUM_BDR_RANDOM	(BRASERO_MEDIUM_BD|		\
					 BRASERO_MEDIUM_WRITABLE|	\
					 BRASERO_MEDIUM_RANDOM)
/* This is always RANDOM write */
#define BRASERO_MEDIUM_BDRE		(BRASERO_MEDIUM_BD|			\
					 BRASERO_MEDIUM_REWRITABLE)



#define BRASERO_MEDIUM_VALID(media)	((media) != BRASERO_MEDIUM_NONE	&&	\
					 (media) != BRASERO_MEDIUM_BUSY	&&	\
					 (media) != BRASERO_MEDIUM_UNSUPPORTED)


#define BRASERO_MEDIUM_TYPE(media)	((media) & 0x003F)
#define BRASERO_MEDIUM_ATTR(media)	((media) & 0x1C000)
#define BRASERO_MEDIUM_STATUS(media)	((media) & 0xE0000)
#define BRASERO_MEDIUM_SUBTYPE(media)	((media) & 0x1FC0)
#define BRASERO_MEDIUM_INFO(media)	((media) & 0xFE0000)

#define BRASERO_MEDIUM_IS(media, type)	(((media)&(type))==(type))

/* These behave like random writable:
 * they can be appendable or closed, rewritable or writable */
#define BRASERO_MEDIUM_RANDOM_WRITABLE(media)					\
	(BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_DVDRW_PLUS) ||		\
	 BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_DVDRW_RESTRICTED) ||		\
	 BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_DVD_RAM) || 			\
	 BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_BDR_RANDOM) ||		\
	 BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_BDR_SRM_POW) ||		\
	 BRASERO_MEDIUM_IS (media, BRASERO_MEDIUM_BDRE))


G_END_DECLS

#endif /* _BURN_MEDIA_H */

 

